.github/workflows/main.yml

```diff
@@ -0,0 +1,272 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "master"
+      - "github-actions" # TODO: remove after build is successful
+  pull_request:
+    branches:
+      - "*"
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  # go needs absolute directories, using the $HOME variable doesn't work here.
+  GOCACHE: /home/runner/work/go/pkg/build
+  GOPATH: /home/runner/work/go
+  DOWNLOAD_CACHE: /home/runner/work/download_cache
+  BITCOIN_VERSION: 0.19.1
+  GO_VERSION: 1.13.10
+
+jobs:
+  ########################
+  # RPC compilation check
+  ########################
+  rpc-check:
+    name: RPC compilation check
+    runs-on: ubuntu-latest
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v1
+
+      - name: go cache
+        uses: actions/cache@v1
+        with:
+          path: /home/runner/work/go
+          key: lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-
+            lnd-${{ runner.os }}-go-
+
+      - name: setup go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~${{ env.GO_VERSION }}'
+
+      - name: download cache
+        uses: actions/cache@v1
+        with:
+          path: /home/runner/work/download_cache
+          key: lnd-${{ runner.os }}-download-${{ hashFiles('**/install_travis_proto.sh') }}
+          restore-keys: |
+            lnd-${{ runner.os }}-download-${{ hashFiles('**/install_travis_proto.sh') }}
+            lnd-${{ runner.os }}-download-
+
+      - name: install protoc and protobuf libraries
+        run: ./scripts/install_travis_proto.sh
+
+      - name: run check
+        run: make rpc-check
+
+  ########################
+  # compile unit tests
+  ########################
+  unit-compile:
+    name: compile unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v1
+
+      - name: go cache
+        uses: actions/cache@v1
+        with:
+          path: /home/runner/work/go
+          key: lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            lnd-${{ runner.os }}-go-${{ env.GO_VERSION }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
```

Comment:
Not sure if this possible but perhaps the restore-keys value could be made with a template? (Seems to repeat on every job).

Reply:
I would've liked to do that. But from what I understand, there isn't any templating option available yet that works within the same file. We could eventually extract these steps into their own actions, but that seemed a bit overkill for just getting the GitHub Workflows to work.